python/pyarrow/table.pxi

```diff
@@ -1407,6 +1407,28 @@ cdef class ChunkedArray(_PandasConvertible):
         self.init(c_chunked_array)
         return self
 
+    @property
+    def device_type(self):
+        """
+        The device type where the chunks in the ChunkedArray reside.
+
+        Returns
+        -------
+        DeviceAllocationType
+        """
+        return _wrap_device_allocation_type(self.sp_chunked_array.get().device_type())
+
+    @property
+    def is_cpu(self):
```

Comment:
An `is_cpu` predicate on chunked arrays can be defined without us forcing a single device type for chunked arrays. This would unblock the Python checks without ruling out the possibility of arrays with mixed device allocations.